packages/auth/amplify_auth_cognito/lib/src/auth_plugin_impl.dart

```diff
@@ -357,6 +365,88 @@ class AmplifyAuthCognito extends AuthPluginInterface implements Closeable {
     return CognitoResendSignUpCodeResult(codeDeliveryDetails);
   }
 
+  @override
+  Future<SignOutResult> signOut({
+    SignOutRequest? request,
+  }) async {
+    final options = request?.options ?? const SignOutOptions();
+
+    // Try to retrieve tokens and return successfully if already logged out.
+    // Do not clear other storage items (e.g. AWS credentials) in this case,
+    // since an unauthenticated user may still be cached.
+    final CognitoUserPoolTokens tokens;
```

Comment:
Will need to change this logic, obviously, when we implement identity pool federation; but this works fine for now.